mlflow/pyfunc/__init__.py

```diff
@@ -3140,6 +3140,8 @@ def predict(self, context, model_input: List[str], params=None) -> List[str]:
                 not _signature_cannot_be_inferred_from_type_hint(type_hints.input)
             )
             if should_infer_signature_from_type_hints:
+                context = PythonModelContext(artifacts, model_config)
+                model_for_signature_inference.load_context(context)
```

Comment:
Thanks, this should work but could you move this up to line 3133? Since this part applies to callables as well.
BTW could you add your sample as a test case so we can make sure this fixes the issue?

Reply:
BTW the current code doesn't pass context to the predict function, but you don't use 'context' parameter in your predict function directly so your case should be fine.